server/src/main/scala/org/http4s/server/middleware/Timeout.scala

```diff
@@ -33,15 +37,18 @@ object Timeout {
 
   /** Transform the service to return a RequestTimeOut [[Status]] after the supplied Duration
     * @param timeout Duration to wait before returning the RequestTimeOut
+    * @param customScheduler a custom scheduler to use for timing out the request.
+    *   If `None` is provided, then a default daemon thread scheduler will be used.
     * @param service [[HttpService]] to transform
     */
-  def apply(timeout: Duration, response: Task[Response] = DefaultTimeoutResponse)(service: HttpService): HttpService = {
-    if (timeout.isFinite()) apply(timeoutResp(timeout, response))(service)
+  def apply(timeout: Duration, response: Task[Response] = DefaultTimeoutResponse, customScheduler: Option[ScheduledExecutorService] = None)(service: HttpService): HttpService = {
```

Comment:
Yeah, I guess this is nicer.  We handle it sensibly, and if a user provides a configuration typed as `Duration`, they don't have to pattern match it themselves this way.  Let's leave it alone.